calculator_test.go

```diff
@@ -2,23 +2,127 @@ package calculator_test
+		{4323, 12, 4311, "Subtracting a positive number from a larger positive number yields a positive number"},
+		{11.2, 89.8, -78.6, "Subtracting a positive number from a smaller positive number yields a negative number"},
+		{-100, -111, 11, "Subtracting a negative number from a larger negative number yields a positive number"},
+		{0, 6, -6, "Subtracting a positive number from zero yields a negative number"},
+		{0, -22, 22, "Subtracting a negative number from zero yields a positive number"},
+	}
+	for _, c := range testCases {
+		got := calculator.Subtract(c.a, c.b)
+		if c.want != got {
+			t.Errorf("%s: want %f, got %f", c.name, c.want, got)
+		}
 	}
 }
 
-// func TestSubtract(t *testing.T) {
-// 	t.Parallel()
-// 	var want float64 = 2
-// 	got := calculator.Subtract(4, 2)
-// 	if want != got {
-// 		t.Errorf("want %f, got %f", want, got)
-// 	}
-// }
+func TestMultiply(t *testing.T) {
+	t.Parallel()
+	var testCases = []struct {
+		a, b, want float64
+		name       string
+	}{
+		{1, 1, 1, "Multiplying one by one always yields one"},
+		{3999191, 1, 3999191, "Multiplying a number by one yields that number"},
+		{3, 3, 9, "Multiplying a number by itself equals that number squared"},
+		{-10, -10, 100, "Multiplying two negative number yields a positive number"},
+		{12, -1.5, -18, "Multiplying a negative number by a positive number yields a negative number"},
+		{43, 0, 0, "Multiplying any number by zero yields zero"},
+	}
+	for _, c := range testCases {
+		got := calculator.Multiply(c.a, c.b)
+		if c.want != got {
+			t.Errorf("%s: want %f, got %f", c.name, c.want, got)
+		}
+	}
+
+}
+
+func TestDivide(t *testing.T) {
+	t.Parallel()
+	var testCases = []struct {
+		a, b, want    float64
+		errorExpected bool
+		name          string
+	}{
+		{10, 1, 10, false, "Dividing a number by 1 yields that number"},
+		{312, 0, 0, true, "Dividing a number by zero returns an error"},
+		{0, 0, 0, true, "Dividing a zero by zero returns an error"},
+		{1, 2, 0.5, false, "Dividing a two numbers may yield a fraction"},
+		{100, 10000000, 0.00001, false, "Dividing a number by a much larger number yields a small fraction"},
+	}
+	for _, c := range testCases {
+		got, err := calculator.Divide(c.a, c.b)
+		if c.errorExpected && err == nil {
+			t.Errorf("%s: expected error, got nil", c.name)
+		}
+		if !c.errorExpected && err != nil {
+			t.Errorf("%s: expected no errors, got \"%v\"", c.name, err)
+		}
+		if c.want != got {
+			t.Errorf("%s: want %f, got %f", c.name, c.want, got)
+		}
+	}
+}
+
+func TestSqrt(t *testing.T) {
+	t.Parallel()
+	var testCases = []struct {
+		a, want       float64
+		errorExpected bool
+		name          string
+	}{
+		{1, 1, false, "The square root of one is one"},
+		{0, 0, false, "The square root of zero is zero"},
+		{25, 5, false, "The square root of a two digit number"},
+		{1947028, math.Sqrt(1947028), false, "The square root of a large number"},
+		{0.00000002, math.Sqrt(0.00000002), false, "The square root of a small fraction"},
```

Comment:
Is it generally a good practice to rely on the standard lib accuracy/correctness in tests or is it better to be more explicit with what's expected here?

Reply:
In general, wherever possible we should pre-calculate the expected answer in some out-of-band way (whether that's the Go standard library, a calculator, pencil and paper, or whatever). So I would get the results of these calls, and hard-wire the exact floating-point value into the test as a constant literal.

Reply:
Done.